controllers/k8ssandra/medusa_reconciler.go

```diff
@@ -37,16 +39,34 @@ func (r *K8ssandraClusterReconciler) reconcileMedusa(
 				return result.Error(fmt.Errorf("medusa encryption certificates were not provided despite client encryption being enabled"))
 			}
 		}
-		if medusaSpec.StorageProperties.StorageProvider != "local" && medusaSpec.StorageProperties.StorageSecretRef.Name == "" {
+		if medusaSpec.StorageProperties.StorageSecretRef.Name == "" {
 			return result.Error(fmt.Errorf("medusa storage secret is not defined for storage provider %s", medusaSpec.StorageProperties.StorageProvider))
 		}
 		if res := r.reconcileMedusaConfigMap(ctx, remoteClient, kc, logger, namespace); res.Completed() {
 			return res
 		}
 
+		medusaContainer, err := medusa.CreateMedusaMainContainer(dcConfig, medusaSpec, kc.Spec.UseExternalSecrets(), kc.SanitizedName(), logger)
+		if err != nil {
+			return result.Error(err)
+		}
 		medusa.UpdateMedusaInitContainer(dcConfig, medusaSpec, kc.Spec.UseExternalSecrets(), kc.SanitizedName(), logger)
-		medusa.UpdateMedusaMainContainer(dcConfig, medusaSpec, kc.Spec.UseExternalSecrets(), kc.SanitizedName(), logger)
-		medusa.UpdateMedusaVolumes(dcConfig, medusaSpec, kc.SanitizedName())
+		medusa.UpdateMedusaMainContainer(dcConfig, medusaContainer)
+
+		// Create required volumes for the Medusa containers
+		volumes := medusa.GenerateMedusaVolumes(dcConfig, medusaSpec, kc.SanitizedName())
+		for _, volume := range volumes {
+			cassandra.AddOrUpdateVolume(dcConfig, volume.Volume, volume.VolumeIndex, volume.Exists)
+		}
+
+		// Create the Medusa standalone pod
+		desiredMedusaStandalone := medusa.StandaloneMedusaDeployment(medusaContainer, kc.Name, dcConfig.Meta.Name, namespace, logger)
+
+		// Add the volumes previously computed to the Medusa standalone pod
+		for _, volume := range volumes {
+			cassandra.AddOrUpdateVolumeToSpec(&desiredMedusaStandalone.Spec.Template, volume.Volume, volume.VolumeIndex, volume.Exists)
```

Comment:
Issue (might have to be a suggestion): At least for this one (given that it isn't a Cassandra pod) can we please use Coalesce? 

This is tough to interpret for anyone who comes after us, because it is using logic originally developed for manipulating Cassandra pod's podTemplateSpecs in a totally different context. 

It also requires an additional for loop which goalesce would abstract away. 

I don't think I can actually block this PR on those grounds, but I do feel somewhat strongly about this one. Perhaps it needs to be a more wholistic review of the codebase with a view to removing this function completely and replacing it with Goalesce throughout all projects.